File: Lab23/LambdaFunctions.py
max_value = lambda numbers: max(numbers)

squares = lambda numbers: list(map(lambda x: x**2, numbers))

sum_of_squares = lambda numbers: sum(map(lambda x: x**2, numbers))

average = lambda numbers: sum(numbers) / len(numbers) if len(numbers) > 0 else 0

import math
circle_area = lambda r: math.pi * r**2

filter_with_a = lambda words: list(filter(lambda w: 'a' in w, words))

f_to_c = lambda f: (f - 32) * 5/9

numbers = [3, 7, 2, 9, 4]
words = ["apple", "pear", "grape", "melon", "berry"]

print("1. Max value:", max_value(numbers))
print("2. Squares:", squares(numbers))
print("3. Sum of squares:", sum_of_squares(numbers))
print("4. Average:", average(numbers))
print("5. Circle area (r=5):", circle_area(5))
print("6. Filter words with 'a':", filter_with_a(words))
print("7. 98°F in Celsius:", f_to_c(98))

File: Lab23/LambdaFunctions.kt
import kotlin.math.PI
import kotlin.math.pow

fun main() {
    val maxValue = { numbers: List<Int> -> numbers.max() }

    val squares = { numbers: List<Int> -> numbers.map { it * it } }

    val sumOfSquares = { numbers: List<Int> -> numbers.map { it * it }.sum() }

    val average = { numbers: List<Int> ->
        if (numbers.isNotEmpty()) numbers.sum().toDouble() / numbers.size else 0.0
    }

    val circleArea = { r: Double -> PI * r.pow(2) }

    val filterWithA = { words: List<String> -> words.filter { "a" in it } }

    val fToC = { f: Double -> (f - 32) * 5 / 9 }

    val numbers = listOf(3, 7, 2, 9, 4)
    val words = listOf("apple", "pear", "grape", "melon", "berry")

    println("1. Max value: ${maxValue(numbers)}")
    println("2. Squares: ${squares(numbers)}")
    println("3. Sum of squares: ${sumOfSquares(numbers)}")
    println("4. Average: ${average(numbers)}")
    println("5. Circle area (r=5): ${circleArea(5.0)}")
    println("6. Filter words with 'a': ${filterWithA(words)}")
    println("7. 98°F in Celsius: ${fToC(98.0)}")
}